satpy/tests/reader_tests/test_fci_l1c_fdhsi.py

```diff
@@ -0,0 +1,380 @@
+                "ir_{:>02d}": (38, 87, 97, 105, 123, 133),
+                "wv_{:>02d}": (63, 73),
+                }
+        data = {}
+        for pat in chan_patterns.keys():
+            for ch_num in chan_patterns[pat]:
+                data.update(self._get_test_content_for_channel(pat, ch_num))
+        return data
+
+    def _get_test_content_areadef(self):
+        data = {}
+        proc = "state/processor"
+        for (lb, no) in (
+                ("earth_equatorial_radius", 6378137),
+                ("earth_polar_radius", 6356752),
+                ("reference_altitude", 35786000),
+                ("projection_origin_longitude", 0)):
+            data[proc + "/" + lb] = xr.DataArray(no)
+        return data
+
+    def get_test_content(self, filename, filename_info, filetype_info):
+        # mock global attributes
+        # - root groups global
+        # - other groups global
+        # mock data variables
+        # mock dimensions
+        #
+        # ... but only what satpy is using ...
+
+        D = {}
+        D.update(self._get_test_content_all_channels())
+        D.update(self._get_test_content_areadef())
+        return D
+
+
+class FakeNetCDF4FileHandler3(FakeNetCDF4FileHandler2):
+    """Mock bad data
+    """
+    def _get_test_calib_for_channel_ir(self, chroot, meas):
+        from netCDF4 import default_fillvals
+        v = xr.DataArray(default_fillvals["f4"])
+        data = {}
+        data[meas + "/radiance_unit_conversion_coefficient"] = v
+        data[chroot + "/central_wavelength_actual"] = v
+        data[meas + "/radiance_to_bt_conversion_coefficient_a"] = v
+        data[meas + "/radiance_to_bt_conversion_coefficient_b"] = v
+        data[meas + "/radiance_to_bt_conversion_constant_c1"] = v
+        data[meas + "/radiance_to_bt_conversion_constant_c2"] = v
+        return data
+
+
+class TestFCIL1CFDHSIReader(unittest.TestCase):
+    yaml_file = "fci_l1c_fdhsi.yaml"
+
+    _alt_handler = FakeNetCDF4FileHandler2
+
+    def setUp(self):
+        """Wrap NetCDF4 FileHandler with our own fake handler
+        """
+
+        # implementation strongly inspired by test_viirs_l1b.py
+        from satpy.config import config_search_paths
+        from satpy.readers.fci_l1c_fdhsi import FCIFDHSIFileHandler
+
+        self.reader_configs = config_search_paths(
+                os.path.join("readers", self.yaml_file))
+        self.p = mock.patch.object(
+                FCIFDHSIFileHandler,
+                "__bases__",
+                (self._alt_handler,))
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def tearDown(self):
+        """Stop wrapping the NetCDF4 file handler
+        """
+        # implementation strongly inspired by test_viirs_l1b.py
+        self.p.stop()
+
+
+class TestFCIL1CFDHSIReaderGoodData(TestFCIL1CFDHSIReader):
+    """Test FCI L1C FDHSI reader
+    """
+
+    # TODO:
+    # - test special case for extended range IR38
+    # - test geolocation
```

Comment:
Are these still needed?

Reply:
Yes, in fact those TODOs belong not only to the tests but also to the reader.  The 3.8 µm channel has a special mode with a "warm_scale_factor" and a "warm_add_offset", designed for fire detection.  This is not included in the current test dataset, so both the implementation and their test remains to be done.

The geolocation testing is still very limited.  I still need to plot the data with a coastline on top to at least visually inspect it, I'm not sure how to to test it, but currently `get_area_def` is not explicitly tested.  It's apparently covered because it gets called when loading data but I'm not checking the output in any way yet.